### PR TITLE
Allow setting a different region for S3 upload

### PR DESCRIPTION
**Pull request by chrisgilmerproj**
It would be nice if uploading to S3 could be set to a different region than the stack that is being created. This adds a single option to so so with either `-s` or `--s3-region` and maintains backwards compatibility.

### CODE REVIEW

File: scripts/cfn [hunk 1]
```diff
@@ -131,7 +134,11 @@ if __name__ == "__main__":
 
         if values.s3bucket:
             # Upload to S3 and create the stack
-            s3conn = boto.s3.connect_to_region(values.region)
+            s3conn = None
```
Comment (markpeek) [inline scripts/cfn]: I don't believe this is needed.

File: scripts/cfn [hunk 2]
```diff
@@ -131,7 +134,11 @@ if __name__ == "__main__":
...
+            if values.s3region != values.region:
```
Comment (markpeek) [inline scripts/cfn]: Would this be better stated as ```if values.s3region is not None:```?

File: scripts/cfn (hunk 1 above)
Comment (chrisgilmerproj) [inline scripts/cfn]: My linter complained :( It's not necessary but its apparently more correct.  Let me know if you'd like me to remove it.

File: scripts/cfn (hunk 2 above)
Comment (chrisgilmerproj) [inline scripts/cfn]: I set the default to be `us-east-1`.  Thinking about it, I could just forgo this entire statement and just change the original to:

```py
s3conn = boto.s3.connect_to_region(values.s3region)
```

What do you think?  If you'd prefer it to be `None` as a default I can update that as well.

**Comment by chrisgilmerproj** [comment]
Take a look now, I think this clears it up.

**Comment by markpeek** [comment]
The concern I had was existing usage of the command where the s3region was assumed to be the same as the region. With your change you have to set both. It might be better to default s3region to None in the  add_argument() call and then set s3region to region if s3region is None. In other words, I liked your previous version better but had just commented on slight tweaks to it. Thanks!

**Comment by chrisgilmerproj** [comment]
@markpeek - I see what you we're saying.  I've made the change you suggested and also updated the help text to read correctly.

I merged this with the latest master so it ought to pass tests.  If you need me to rebase it into one commit please let me know but I think its ready for your sign off.

**Comment by markpeek** [comment]
Thanks!

**Comment by chrisgilmerproj** [comment]
Thank you!

**Pull request merged**